packages/javascript-sdk/src/fr-webauthn/index.ts

```diff
@@ -159,13 +159,19 @@ abstract class FRWebAuthn {
         if (metadataCallback) {
           const meta = metadataCallback.getOutputValue('data') as WebAuthnRegistrationMetadata;
           publicKey = this.createRegistrationPublicKey(meta);
+          const credential = await this.getRegistrationCredential(
+            publicKey as PublicKeyCredentialCreationOptions,
+          );
+          outcome = this.getRegistrationOutcome(credential);
         } else if (textOutputCallback) {
           publicKey = parseWebAuthnRegisterText(textOutputCallback.getMessage());
+          const credential = await this.getRegistrationCredential(
+            publicKey as PublicKeyCredentialCreationOptions,
+          );
+          outcome = this.getRegistrationOutcome(credential);
+        } else {
+          throw new Error('No Credential found from Public Key');
```

Comment:
Here is some typescript related changes i made, and we would technically need to regress this.

Reply:
Can we just pull these duplicated code into a function using generics? Also, since we're pulling so much into the conditional blocks, should we just pull it all up into the function and just return in the block?

Reply:
We can totally refactor this. let me do that then, i was just keeping the surface area minimal when i went through this.

Reply:
im looking at this and im not sure how duplicative this really is, do you have something in mind?

Reply:
we want to extract two lines into its own function?

Reply:
Something like this, but we can discuss this some other time:

```js
async function registerAndSetInput(publicKey) {
  const credential = await this.getRegistrationCredential(
    publicKey as PublicKeyCredentialCreationOptions,
  );
  const outcome = this.getRegistrationOutcome(credential);
  hiddenCallback.setInputValue(
    deviceName && deviceName.length > 0 ? `${outcome}::${deviceName}` : outcome,
  );
}
if (metadataCallback) {
  const meta = metadataCallback.getOutputValue('data') as WebAuthnRegistrationMetadata;
  publicKey = this.createRegistrationPublicKey(meta);
  await registerAndSetInput(publicKey);
  return step; // return in conditional block, rather than outside
} else if (textOutputCallback) {
  publicKey = parseWebAuthnRegisterText(textOutputCallback.getMessage());
  await registerAndSetInput(publicKey);
  return step;
} else {
  // error case ...
```